pkg/rpcserver/rpcserver.go

```diff
@@ -416,7 +416,9 @@ func (serv *server) CreateInstance(id int, injectExec chan<- bool, updInfo dispa
 		infoc:         make(chan chan []byte),
 		requests:      make(map[int64]*queue.Request),
 		executing:     make(map[int64]bool),
-		lastExec:      MakeLastExecuting(serv.cfg.Procs, 6),
+		hanged:        make(map[int64]bool),
+		// Executor may report proc IDs that are larger than serv.cfg.Procs.
+		lastExec:      MakeLastExecuting(prog.MaxPids, 6),
```

Comment:
Nit: I wonder if we should additionally take the last 30 of `Collect()` results (or do it right inside `Collect()`) -- to additionally restrict the size of the resulting log.

Reply:
We can't just take 30. If there is a slow test, it can be the last test on a proc (must not remove), but also can be almost arbitrary old.

Reply:
Also all hanged programs are old.

Reply:
If it's 40-60 programs, it likely just won't fit into the 2MB compressed log size we share from the dashboard. Especially if there are `syz_mount_image` calls in there.

Reply:
That's why I wanted to keep 2 ids in executor, it does not lead to log size increase (beyond hanged programs).